docs/src/manual/nonlinear.md

```diff
@@ -622,53 +626,166 @@ f_splat(x..., y..., z)
 @objective(model, Min, op_f(x..., y..., z))
 ```
 
-### Automatic differentiation
+### Common mistakes when writing a user-defined operator
 
-JuMP does not support black-box optimization, so all user-defined operators must
-provide derivatives in some form. Fortunately, JuMP supports automatic
-differentiation of user-defined operators.
+When nonlinear expressions are provided algebraically (either via direct input
+into the macros or via [Function tracing](@ref)), JuMP computes first- and
+second-order derivatives using sparse reverse-mode automatic differentiation.
+For details, see [ReverseAD](@ref).
 
-!!! info
-    Automatic differentiation is *not* finite differencing. JuMP's automatically
-    computed derivatives are not subject to approximation error.
+However, because [ReverseAD](@ref) requires the algebraic expression as input,
+JuMP cannot use [ReverseAD](@ref) to differentiate user-defined operators.
 
+Instead, unless you provide your own derivatives with [Gradients and Hessians](@ref),
 JuMP uses [ForwardDiff.jl](https://github.com/JuliaDiff/ForwardDiff.jl) to
-perform automatic differentiation of user-defined operators; see the ForwardDiff.jl
-[documentation](https://www.juliadiff.org/ForwardDiff.jl/v0.10.2/user/limitations.html)
-for a description of how to write a function suitable for automatic
-differentiation.
+compute derivatives.
 
-#### Common mistakes when writing a user-defined operator
+ForwardDiff has a number of limitations that you should be aware of when
```

Comment:
Maybe wise to mention that gradients are very slow in high dimension cause forward mode AD?